packages/server/README.md

```diff
@@ -22,3 +22,158 @@ To build the NATIVE image simply run the following from the root directory.
+
+How the `@JoinColumn` annotation is used is based on the type of relationship, as per the docs:
+
+```html
+/*
+* - If the join is for a OneToOne or ManyToOne
+*  mapping using a foreign key mapping strategy,
+* the foreign key column is in the table of the
+* source entity or embeddable.
+
+* - If the join is for a unidirectional OneToMany mapping
+* using a foreign key mapping strategy, the foreign key is in the
+* table of the target entity.
+
+* - If the join is for a ManyToMany mapping or for a OneToOne
+* or bidirectional ManyToOne/OneToMany mapping using a join
+* table, the foreign key is in a join table.
+
+* -  If the join is for an element collection, the foreign
+* key is in a collection table.
+*/
+```
+
+```java
+@Data
+@Entity
+class Parent {
+
+    @Id
+    private UUID id = UUID.randomUUID();
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+	@JoinColumn(name = "parent_id") // <--- because this is a @OneToMany uni-directional relationship, its referencing the column on the child, not the parent.
+    private List<Child> children = new ArrayList<>();
+
+};
+
+@Data
+@Entity
+class Child {
+
+    @Id
+    private UUID id = UUID.randomUUID();
+
+};
+```
+
+## Migrations:
+
+There are two "passes" for migrations in the app.
+
+1) For local, staging, and prod we have `spring.jpa.hibernate.ddl-auto=update`. On server startup Hibernate will update the schema if it detects that the entities and tables/columns are out of sync. It will add new tables/columns and update existing ones as needed, but it won't delete any existing tables or columns.
+
+This ensures that new tables are created and new columns are added. However this leaves the case of _modifying existing columns_.
+
+2) To modify existing columns we leverage [Flyway](https://github.com/flyway/flyway) which executes _after_ the hibernate DDL pass. There are two types of flyway migrations, `.sql` migrations, and Java migrations.
+
+### .sql migrations:
+
+To add a `.sql` migration, add the migraiton file to the `resources/db/migration` dir, ensure the migration fits the naming convention: `V{VERSION_NUM}__Name_of_migration.sql`, where `VERSION_NUM` is a new value incremented from the previous latest version.
+
+**These are the preferred method of migrating**
+
+### .java migrations:
+
+To add a `.java` migration to the `packages/server/src/main/java/db/migration` dir in the following form:
+
+```java
+package db.migration;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+
+public class V8__Some_java_migration extends BaseJavaMigration {
+    @Override
+    public void migrate(Context context) throws Exception {
+        try (var statement = context.getConnection().createStatement()) {
+            statement.execute("ALTER TABLE simulation ALTER COLUMN description TYPE text;");
+        }
+    }
+}
+```
+
+Only use a java based migration if it is not possible to achieve via a `.sql` one.
```

Comment:
This is great!  It sheds light on a grossly obsfucating pointless layer of bad code design.